algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+        return Value(Type.make_dynamic_array_type(element_type), values)
+
+    @staticmethod
+    def make_static_array(values):
+        assert isinstance(values, list())
+        if len(values) >= (2 ** 16) or len(values) == 0:
+            raise error.ABIValueError(
+                f"dynamic array values length: {len(values)} must be greater than 0 and less than uint16"
+            )
+        for val in values:
+            assert isinstance(val, Value)
+            if val.abi_type != values[0].abi_type:
+                raise error.ABIValueError(
+                    f"dynamic array type mismatch: {val.abi_type} != {values[0].abi_type}"
+                )
+        return Value(
+            Type.make_static_array_type(values[0].abi_type, len(values)),
+            values,
+        )
+
+    @staticmethod
+    def make_tuple(values):
+        assert isinstance(values, list())
+        if len(values) >= (2 ** 16):
+            raise error.ABIValueError(
+                f"tuple values length: {len(values)} must be less than uint16"
+            )
+        tuple_types = []
+        for val in values:
+            assert isinstance(val, Value)
+            tuple_types.append(val.abi_type)
+
+        return Value(Type.make_tuple_type(tuple_types), values)
+
+    def get_uint(self):
+        if not isinstance(self.abi_type.abi_type_id, BaseType.Uint):
+            raise error.ABIValueError(
+                f"value type {self.abi_type.abi_type_id} is not a uint"
+            )
+        if self.value >= (2 ** self.abi_type.bit_size):
+            raise error.ABIValueError(
+                f"value {self.value} is too big to fit in size {self.abi_type.bit_size}"
+            )
+        return self.value
+
+    def get_uint8(self):
+        if (
+            self.abi_type.abi_type_id != BaseType.Uint
+            or self.abi_type.bit_size != 8
+        ):
+            raise error.ABIValueError(
+                f"value type {self.abi_type.abi_type_id} is not a uint8"
+            )
+        return self.get_uint()
+
+    def get_uint16(self):
+        if (
+            self.abi_type.abi_type_id != BaseType.Uint
+            or self.abi_type.bit_size != 16
+        ):
+            raise error.ABIValueError(
+                f"value type {self.abi_type.abi_type_id} is not a uint16"
+            )
+        return self.get_uint()
+
+    def get_uint32(self):
+        if (
+            self.abi_type.abi_type_id != BaseType.Uint
+            or self.abi_type.bit_size != 32
+        ):
+            raise error.ABIValueError(
+                f"value type {self.abi_type.abi_type_id} is not a uint32"
+            )
+        return self.get_uint()
+
+    def get_uint64(self):
+        if (
+            self.abi_type.abi_type_id != BaseType.Uint
+            or self.abi_type.bit_size != 64
+        ):
+            raise error.ABIValueError(
+                f"value type {self.abi_type.abi_type_id} is not a uint64"
+            )
+        return self.get_uint()
```

Comment:
I don't think these functions are useful in python, since there is only one int type and it's essentially a bigint, so it can hold all values.